### PR TITLE
test: Don't use GPL module when Base.USE_GPL_LIBS=false

### PR DESCRIPTION
**Pull request by inkydragon**
The GPL module `SparseArrays.CHOLMOD` has already been used before the `if Base.USE_GPL_LIBS` statement.

This results in the error
```
ERROR: LoadError: UndefVarError: `CHOLMOD` not defined in `SparseArrays`
Stacktrace:
```

### CODE REVIEW

**Comment by gitboy16** [comment]
Is my understanding correct if I say that tests are failing because of method ambiguities between SparseArrays and LinearAlgebra when Aqua.jl runs?

**Comment by ViralBShah** [comment]
Should we disable the aqua tests until they run more reliably?

**Comment by inkydragon** [comment]
> disable the aqua tests

Or run Aqua tests in a standalone CI job

**Comment by ViralBShah** [comment]
Is this ok to merge? I assume we need to fix the Aqua tests separately (and those failures are not related to this PR).

File: test/spqr.jl [hunk 1]
```diff
@@ -10,7 +14,7 @@ using SparseArrays: SparseArrays, sparse, sprandn, spzeros, SparseMatrixCSC
 using Random: seed!
 
 # TODO REMOVE SECOND PREDICATE WITH SS7.1
```
[the remark this answers is not in the document]
Comment (SobhanMP) [reply test/spqr.jl:16]: Can you also remove the comment as well?

**Comment by SobhanMP** [comment]
I don't think anything can cause failure, but it might be a good to also run the tests with USE_GPL_LIB=false...

**Comment by dkarrasch** [comment]
I believe the Aqua tests were running reliably. A commit on Julia master introduced method ambiguities that have been resolved meanwhile. They were failing even on the "split Aqua tests" PR. ;-)

File: test/cholmod.jl [hunk 2]
```diff
@@ -1,8 +1,12 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
 module CHOLMODTests
-
 using Test
+
+@static if !Base.USE_GPL_LIBS
+    @info "Not use GPL libs, Skipping CHOLMOD Tests"
```
Comment (ViralBShah) [inline test/cholmod.jl:7]: Could you update the message to `"This Julia build excludes the use of SuiteSparse GPL libraries. Skipping CHOLMOD tests"`?

Comment (inkydragon) [inline test/cholmod.jl:7]:
```suggestion
    @info "This Julia build excludes the use of SuiteSparse GPL libraries. Skipping CHOLMOD tests"
```

File: test/linalg_solvers.jl [hunk 3]
```diff
@@ -1,14 +1,16 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
 module SparseLinalgSolversTests
-
 using Test
+
+@static if !Base.USE_GPL_LIBS
+    @info "Not use GPL libs, Skipping SparseLinalgSolvers Tests"
```
Comment (inkydragon) [inline test/linalg_solvers.jl:7]:
```suggestion
    @info "This Julia build excludes the use of SuiteSparse GPL libraries. Skipping SparseLinalgSolvers Tests"
```

File: test/spqr.jl [hunk 4]
```diff
@@ -1,16 +1,20 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
 module SPQRTests
-
 using Test
+
+@static if !Base.USE_GPL_LIBS
+    @info "Not use GPL libs, Skipping SPQR Tests"
```
Comment (inkydragon) [inline test/spqr.jl:7]:
```suggestion
    @info "This Julia build excludes the use of SuiteSparse GPL libraries. Skipping SPQR Tests"
```

File: test/umfpack.jl [hunk 5]
```diff
@@ -1,15 +1,19 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
 module UMFPACKTests
-
 using Test
+
+@static if !Base.USE_GPL_LIBS
+    @info "Not use GPL libs, Skipping UMFPACK Tests"
```
Comment (inkydragon) [inline test/umfpack.jl:7]:
```suggestion
    @info "This Julia build excludes the use of SuiteSparse GPL libraries. Skipping UMFPACK Tests"
```

**Review by inkydragon — COMMENTED** [review]
Apply suggestions.

**Pull request merged**